src/commands/lambda/functions/commons.ts

```diff
@@ -171,9 +171,21 @@ export const findLatestLayerVersion = async (layer: LayerKey, region: string) =>
   return latestVersion
 }
 
-export const isMissingAWSCredentials = () =>
-  process.env[AWS_ACCESS_KEY_ID_ENV_VAR] === undefined || process.env[AWS_SECRET_ACCESS_KEY_ENV_VAR] === undefined
-
+export const isMissingAWSCredentials = () => {
+  // If env vars and aws_sdk_config.credentials are not set return true otherwise return false
+  if (
+    process.env[AWS_ACCESS_KEY_ID_ENV_VAR] === undefined ||
+    process.env[AWS_SECRET_ACCESS_KEY_ENV_VAR] === undefined
+  ) {
+    if (aws_sdk_config.credentials) {
+      return false
+    } else {
+      return true
+    }
+  } else {
+    return false
+  }
+}
```

Comment:
This can be simplified to:

```suggestion
export const isMissingAWSCredentials = () => {
  // If env vars and aws_sdk_config.credentials are not set return true otherwise return false
  return
    (process.env[AWS_ACCESS_KEY_ID_ENV_VAR] === undefined ||
    process.env[AWS_SECRET_ACCESS_KEY_ENV_VAR] === undefined) &&
    !aws_sdk_config.credentials
  ) 
}
```

Reply:
ooooo this is way cleaner. Seeing this now makes sense but what sort of tips do you have on simplifying boolean logic like the one I had set earlier?

Reply:
https://github.com/DataDog/datadog-ci/pull/512/commits/80712c3a03f9d87117b62761f770c0c7baf8bc32

Reply:
You should create a truth table with all the inputs and the outputs and then try to simplify the expression into its simplest form that still produces the same outputs for the truth table as the original expression.